src/libServer/LookupServer.cpp

```diff
@@ -1393,18 +1396,24 @@ string LookupServer::GetTotalCoinSupply() {
         AccountStore::GetInstance().GetAccount(NullAddress, true)->GetBalance();
   }
 
-  boost::multiprecision::cpp_dec_float_50 rewards(balance.str());
+  mp::cpp_dec_float_50 rewards(balance.str());
   ans -= rewards;
   ans /= 1000000000000;  // Convert to ZIL
+  return ans;
+}
 
+string LookupServer::GetTotalCoinSupply() {
   ostringstream streamObj;
   streamObj << std::fixed;
   streamObj << std::setprecision(12);
-  streamObj << ans;
-
+  streamObj << CalculateTotalSupply();
   return streamObj.str();
 }
 
+unsigned long LookupServer::GetTotalCoinSupplyAsInt() {
+  return CalculateTotalSupply().convert_to<unsigned long>();
```

Comment:
Will this truncate? Is that what we want?

Reply:
Thank you. It was truncating before. Added rounding logic.

Reply:
We want to round to the nearest integer.